packages/angular/ssr/test/testing-utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { Component } from '@angular/core';
+import { bootstrapApplication } from '@angular/platform-browser';
+import { provideServerRendering } from '@angular/platform-server';
+import { RouterOutlet, provideRouter } from '@angular/router';
+import { setAngularAppManifest } from '../src/manifest';
+
+/**
+ * Sets the Angular application testing manifest for a given locale.
+ * This function configures server-side rendering settings and bootstraps the Angular application for testing.
+ *
+ * @param [locale] - The locale to be used in the base href and displayed in the home component.
+ */
+export function setAngularAppTestingManifest(locale = ''): void {
+  setAngularAppManifest({
+    inlineCriticialCss: false,
+    assets: {
+      'index.server.html': async () =>
+        `
+ <html>
+  <head>
+    <base href="/${locale}/" />
+  </head>
+  <body>
+    <app-root></app-root>
+  </body>
+</html>`,
+    },
+    bootstrap: () => () => {
+      @Component({
+        standalone: true,
+        selector: 'app-root',
+        template: '<router-outlet />',
+        imports: [RouterOutlet],
+      })
+      class AppComponent {}
+
+      @Component({
+        standalone: true,
+        selector: 'app-home',
+        template: `Home works ${locale.toUpperCase()}`,
```

Comment:
Question: Is using `locale` from the closure a meaningful test? Would it be more appropriate to inject `LOCALE_ID` and render that to confirm that Angular was configured correctly?

Reply:
`LOCALE_ID` is injected during the build process by the Angular CLI and not controlled by `@angular/ssr`, so it won't be useful for unit tests. In this case, e2e testing is more appropriate for that case.